src/utils/transactionUtils/transactionUtils.ts

```diff
@@ -326,10 +320,84 @@ export const FieldToInteractionMap: {
 } = {
   name: ANT_INTERACTION_TYPES.SET_NAME,
   ticker: ANT_INTERACTION_TYPES.SET_TICKER,
+  targetID: ANT_INTERACTION_TYPES.SET_TARGET_ID,
   // TODO: add other interactions
 };
 
 export function getInteractionTypeFromField(field: string) {
   // TODO: add contract specification and more interaction fields
   return FieldToInteractionMap[field];
 }
+
+export function mapTransactionDataKeyToPayload(
+  contractType: ContractType,
+  interactionType: AntInteraction | RegistryInteraction,
+  data: string[],
+) {
+  const payload: any = {};
+  try {
+    if (isRegistryInteraction(interactionType)) {
+      if (
+        !isInteractionCompatible({
+          contractType,
+          interactionType,
+          functionName:
+            TRANSACTION_DATA_KEYS[CONTRACT_TYPES.REGISTRY][interactionType]
+              .functionName,
+        })
+      ) {
+        throw new Error(`Interaction is not compatible`);
+      }
+      payload.functionName =
+        TRANSACTION_DATA_KEYS[CONTRACT_TYPES.REGISTRY][
+          interactionType
+        ].functionName;
+      TRANSACTION_DATA_KEYS[CONTRACT_TYPES.REGISTRY][
+        interactionType
+      ].keys.forEach((key, index) => {
+        if (!data[index]) {
+          // if missing transaction data from the url, throw an error
+          throw new Error(
+            `Missing key (${key}) from transaction data in the url. This may be due to the order of the data, the current order is [${data}], the correct order is [${
+              TRANSACTION_DATA_KEYS[CONTRACT_TYPES.REGISTRY][interactionType]
+                .keys
+            }]`,
+          );
+        }
+        payload[key] = data[index];
+      });
+    }
+    if (isAntInteraction(interactionType)) {
+      if (
+        !isInteractionCompatible({
+          contractType,
+          interactionType,
+          functionName:
+            TRANSACTION_DATA_KEYS[CONTRACT_TYPES.ANT][interactionType]
+              .functionName,
+        })
+      ) {
+        throw new Error(`Interaction is not compatible`);
+      }
+      payload.functionName =
+        TRANSACTION_DATA_KEYS[CONTRACT_TYPES.ANT][interactionType].functionName;
+      TRANSACTION_DATA_KEYS[CONTRACT_TYPES.ANT][interactionType].keys.forEach(
+        (key, index) => {
+          if (!data[index]) {
+            // if missing transaction data from the url, throw an error
+            throw new Error(
+              `Missing key (${key}) from transaction data in the url. This may be due to the order of the data, the current order is [${data}], the correct order is [${
+                TRANSACTION_DATA_KEYS[CONTRACT_TYPES.ANT][interactionType].keys
+              }]`,
+            );
+          }
+          payload[key] = data[index];
+        },
+      );
+    }
+  } catch (error) {
+    console.log(error);
```

Comment:
we're swallowing the error here. when it comes to errors we need to either
1. throw them and let the caller decide how to handle
2. emit them to the eventEmitter so they get captured by sentry and displayed

Reply:
i'm of the opinion all of these util functions should throw, and events are emitted from calling `.tsx` components`.

Reply:
ah ok, makes sense, fixing